ludwig/schema/preprocessing.py

```diff
@@ -0,0 +1,751 @@
+        "the dataset.",
+    )
+
+    infer_image_sample_size: Optional[int] = schema_utils.PositiveInteger(
+        default=100,
+        allow_none=False,
+        description="The sample size used for inferring dimensions of images in infer_image_dimensions.",
+    )
+
+    scaling: Optional[str] = schema_utils.StringOptions(
+        ["pixel_normalization", "pixel_standardization"],
+        default="pixel_normalization",
+        allow_none=False,
+        description="The scaling strategy for pixel values in the image.",
+    )
+
+    in_memory: Optional[bool] = schema_utils.Boolean(
+        default=True,
+        description="Defines whether image dataset will reside in memory during the training process or will be "
+        "dynamically fetched from disk (useful for large datasets). In the latter case a training batch "
+        "of input images will be fetched from disk each training iteration.",
+    )
+
+    num_processes: Optional[int] = schema_utils.PositiveInteger(
+        default=1,
+        allow_none=False,
+        description="Specifies the number of processes to run for preprocessing images.",
+    )
+
+
+@dataclass
+@register_preprocessor("audio")
+class AudioPreprocessingConfig(schema_utils.BaseMarshmallowConfig):
+
+    audio_file_length_limit_in_s: Optional[int] = schema_utils.NonNegativeFloat(
+        default=7.5,
+        allow_none=False,
+        description="Float value that defines the maximum limit of the audio file in seconds. All files longer than "
+        "this limit are cut off. All files shorter than this limit are padded with padding_value",
+    )
+
+    missing_value_strategy: Optional[str] = schema_utils.StringOptions(
+        ["fill_with_const", "fill_with_mode", "fill_with_mean", "backfill", "bfill", "pad", "ffill", "drop_row"],
+        default="backfill",
+        allow_none=False,
+        description="What strategy to follow when there's a missing value in an audio column",
+    )
+
+    in_memory: Optional[bool] = schema_utils.Boolean(
+        default=True,
+        description="Defines whether the audio dataset will reside in memory during the training process or will be "
+        "dynamically fetched from disk (useful for large datasets). In the latter case a training batch "
+        "of input audio will be fetched from disk each training iteration.",
+    )
+
+    padding_value: Optional[float] = schema_utils.NonNegativeFloat(
+        default=0.0, allow_none=False, description="Float value that is used for padding."
+    )
+
+    norm: Optional[str] = schema_utils.StringOptions(
+        ["per_file"],
+        default=None,
+        allow_none=True,
+        description="Normalization strategy for the audio files. If None, no normalization is performed. If "
+        "per_file, z-norm is applied on a 'per file' level",
+    )
+
+    audio_feature: Optional[dict] = schema_utils.Dict(
```

Comment:
Hey @connor-mccorm , just a callout here that this is a bit of a beast on the Client. All 4 of these settings have to be set together right? If so, I'm thinking a grouped set of inputs that are all required make the most sense. However, given the way this is defined, we don't have an enum to pull from for `type` and we have to infer that `num_filter_bands` is an integer while `window_*_in_s` are floats.

I bring that all up to say: we have a similar situation with the `missing_value_strategy` of `fill_with_const` and `fill_value`. In those scenarios, we chose to select a reasonable default and have the user specify each setting separately.

I'm curious, what makes `audio_feature` different that we force the customer to specify a Dict instead of individual fields with reasonable defaults?